### PR TITLE
[FLINK-11897][tests] should wait all submitTask methods executed,befo…

### PR DESCRIPTION
**Pull request by chummyhe89**
## What is the purpose of the change

some test cases in `ExecutionGraphSuspendTest` validate no interactions with TaskManager， but they didn't wait until `submitTask` methods had been invoked before resetting counts.

## Brief change log

  - *add a CountDownLatch `submitLatch` in InteractionsCountingTaskManagerGateway*
  - *add a new `InteractionsCountingTaskManagerGateway` constructor with an int parameter to initialize `submitLatch`*
  - *add waitAllTasksSubmitted method to wait until all `submitTask` methods had been invoked*

## Verifying this change
This change is already covered by existing tests, such as

 - *ExecutionGraphSuspendTest.testSuspendedOutOfDeploying*
 - *ExecutionGraphSuspendTest.testSuspendedOutOfRunning*
 - *ExecutionGraphSuspendTest.testSuspendedOutOfFailing*
 - *ExecutionGraphSuspendTest.testSuspendedOutOfFailed*
 - *ExecutionGraphSuspendTest.testSuspendedOutOfCanceling*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/InteractionsCountingTaskManagerGateway.java [hunk 1]
```diff
@@ -32,6 +33,16 @@
 
 	private final AtomicInteger submitTaskCount = new AtomicInteger(0);
 
+	private CountDownLatch submitLatch;
+
+	public InteractionsCountingTaskManagerGateway(){
```
Comment (TisonKun) [inline flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/InteractionsCountingTaskManagerGateway.java]: `) {`

File: flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/InteractionsCountingTaskManagerGateway.java [hunk 2]
```diff
@@ -32,6 +33,16 @@
...
+		submitLatch = new CountDownLatch(0);
+	}
+
+	public InteractionsCountingTaskManagerGateway(final int parallelism){
```
Comment (TisonKun) [inline flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/InteractionsCountingTaskManagerGateway.java]: `) {`

File: flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/InteractionsCountingTaskManagerGateway.java [hunk 3]
```diff
@@ -60,4 +72,12 @@ int getSubmitTaskCount() {
...
 		return cancelTaskCount.get() + submitTaskCount.get();
 	}
+
+	void waitAllTasksSubmitted(){
+		try{
+			submitLatch.await();
+		}catch (InterruptedException e){
+			Thread.currentThread().interrupt();
+		}
+	}
```
Comment (TisonKun) [inline flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/InteractionsCountingTaskManagerGateway.java]: whitespace between braces and other symbol.

**Comment by TisonKun** [comment]
@flinkbot attention @tillrohrmann @zentol

File: flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/InteractionsCountingTaskManagerGateway.java (hunk 3 above)
Comment (chummyhe89) [reply flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/InteractionsCountingTaskManagerGateway.java]: I'm sorry for that, i'll correct it!

File: flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/InteractionsCountingTaskManagerGateway.java (hunk 2 above)
Comment (chummyhe89) [reply flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/InteractionsCountingTaskManagerGateway.java]: I'm sorry for that, i'll correct it!

**Comment by chummyhe89** [comment]
@TisonKun thanks for your review!

**Comment by chummyhe89** [comment]
Please take a look. @tillrohrmann @zentol

**Comment by TisonKun** [comment]
@chummyhe89 I think the community is busy with releasing 1.8 and ASF Project consists of all volunteers. Sorry for the long delay but be aware of that there is no guarantee for reviewing all prs.

**Comment by chummyhe89** [comment]
Yes,i see. thanks for your prompt reply. @TisonKun

**Pull request merged**